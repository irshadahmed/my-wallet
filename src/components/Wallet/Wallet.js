import React, { Component } from 'react';

export class Wallet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      amount: 0,
    };
  }

  /**
   * Update amount
   */
  updateAmount = (event) => {
    this.setState({
      amount: parseInt(event.target.value, 10),
    });
  };

  /**
   * Deposit amount
   */
  depositAmount = () => this.props.depositAmount(this.state.amount);

  /**
   * Withdraw amount
   */
  withdrawAmount = () => this.props.withdrawAmount(this.state.amount);

  /**
   * Render wallet component
   */
  render() {
    return (
      <div className="ui form">
        <div className="inline fields">
          <div className="five wide field">
            <input
              type="text"
              placeholder="Amount"
              className="input-wallet"
              onChange={this.updateAmount}
            />
          </div>

          <div className="five wide field">
            <button
              className="ui primary basic button btn-deposit"
              onClick={this.depositAmount}
            >
              Deposit
            </button>
            <button
              className="ui secondary basic button btn-withdraw"
              onClick={this.withdrawAmount}
            >
              Withdraw
            </button>
          </div>
        </div>
      </div>
    );
  }
}
