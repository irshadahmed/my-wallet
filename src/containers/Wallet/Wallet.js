import { connect } from 'react-redux';

import { depositAmount, withdrawAmount } from '../../actions/balance';
import { Wallet } from '../../components/Wallet/Wallet';

const mapDispatchToProps = {
  depositAmount,
  withdrawAmount,
};

export default connect(null, mapDispatchToProps)(Wallet);
