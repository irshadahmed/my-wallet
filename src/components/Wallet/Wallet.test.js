import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import  { Wallet } from './Wallet';

describe('Wallet', () => {
  let wallet;
  const mockWithdraw = jest.fn();
  const mockDeposit = jest.fn();
  const props = {
    depositAmount: mockDeposit,
    withdrawAmount: mockWithdraw,
  };

  beforeEach(() => {
    wallet = shallow(<Wallet {...props} />)
  });

  it('renders correctly', () => {
    expect(shallowToJson(wallet)).toMatchSnapshot();
  });

  it('creates an input to deposit or withdraw from balance', () => {
    expect(wallet.find('.input-wallet').exists()).toEqual(true);
  });

  describe('when the user types into the wallet input', () => {
    const userBalance = '25';

    beforeEach(() => {
      wallet.find('.input-wallet').simulate('change', {
        target: {
          value: userBalance
        }
      });
    });

    it('updates the local wallet amount in state', () => {
      expect(wallet.state().amount).toEqual(parseInt(userBalance, 10));
    });

    describe('and the user wants to make a deposit', () => {
      beforeEach(() => {
        wallet.find('.btn-deposit').simulate('click');
      });

      it('should dispatch the depositAmount action creator', () => {
        expect(mockDeposit).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });

    describe('and the user wants to make a withdraw', () => {
      beforeEach(() => {
        wallet.find('.btn-withdraw').simulate('click');
      });

      it('should dispatch the withdrawAmount action creator', () => {
        expect(mockWithdraw).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });
  });

});