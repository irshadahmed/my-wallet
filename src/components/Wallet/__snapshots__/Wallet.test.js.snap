// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders correctly 1`] = `
<div
  className="ui form"
>
  <div
    className="inline fields"
  >
    <div
      className="five wide field"
    >
      <input
        className="input-wallet"
        onChange={[Function]}
        placeholder="Amount"
        type="text"
      />
    </div>
    <div
      className="five wide field"
    >
      <button
        className="ui primary basic button btn-deposit"
        onClick={[Function]}
      >
        Deposit
      </button>
      <button
        className="ui secondary basic button btn-withdraw"
        onClick={[Function]}
      >
        Withdraw
      </button>
    </div>
  </div>
</div>
`;
